cmd/url.go

```diff
@@ -50,24 +53,52 @@ The created URL can be used to access the specified component from outside the O
 		checkError(err, "")
 		projectName := project.GetCurrent(client)
 
+		var app string
 		var componentName string
+		var urlName string
+
+		if urlListApplication == "" {
+			app, err = application.GetCurrent(client)
+			checkError(err, "")
+		} else {
+			app = urlListApplication
+		}
+
+		if len(urlListComponent) == 0 {
+			componentName = getComponent(client, "", app, projectName)
+		} else {
+			componentName = urlListComponent
```

Comment:
There is no check if this user provided component exists.
Because of that it is possible to create url in not existing components.
```
▶ odo url create myurl1 --component nonexisting
Adding URL to component: nonexisting
URL created for component: nonexisting

myurl1 - http://myurl1-app-odo.dev.rdu2c.fabric8.io
```